### PR TITLE
Simplify build by moving gemini-dependency version in a property

### PR DESCRIPTION
**Pull request by tgeens**
In this PR the Gemini Blueprint dependency version is extracted into a property `geminiVersion` and is configured in the root-project, together with the Spring version property. This allows the gradle build in the subprojects to be simplified a bit.

### CODE REVIEW

**Comment by tgeens** [comment]
Wait, why did that build locally ? :worried:

**Comment by tgeens** [comment]
Fixed by adding some resolution strategy rules.

I'm not sure it's still worth it. Opinions ?

File: buildSrc/src/main/java/eu/xenit/dynamicextensionsalfresco/Versions.java [hunk 1]
```diff
@@ -24,6 +24,13 @@
         put("60", "5.0.4.RELEASE");
     }};
 
+    public static final Map<String, String> geminiVersions = new HashMap<String, String>() {{
+        put("50", "1.0.2.RELEASE");
```
Comment (kerkhofsd) [inline buildSrc/src/main/java/eu/xenit/dynamicextensionsalfresco/Versions.java]: Is this still required? The version will be overwritten by the resolution strategy anyway?

**Comment by kerkhofsd** [comment]
@tgeens 
No strong opinion. 
- I am a bit worried about the 'magic' of the resolution strategy. IMO it handles the Gemini dependency change more 'behind the scenes', where before this was really explicitly visible for each subproject. 
+ I do like the simplification of the other build files

**Comment by vierbergenlars** [comment]
Using dependency substitution like that has the potential to be very confusing, especially since the substitution is controlled by an other project property.

I would like the dependency substitution rule to switch the coordinates based on the version of the requested dependency.

I can agree with the idea that subprojects should not care about the exact coordinates of a dependency if they have changed through history, but actually refer to the same logical package.
But now, the dependency configuration contains a dependency that does not exist anywhere. You have to go look at a second place in the code base to figure out that you are actually rewriting the dependency.

I think I prefer the explicitness of keeping the changing dependencies in a separate gradle file that is explicitly included.

File: build.gradle [hunk 2]
```diff
@@ -201,6 +203,33 @@ def allSubProjectsConfig = {
                         details.because "'org.springframework.extensions.surf' was moved to 'org.alfresco.surf'"
                     }
                 }
+
+                // Eclipse Gemini Blueprint 1.0.2.RELEASE was published as 'org.eclipse.gemini' and is
+                // the last version that supports Spring 3.x
+                // Starting from 2.0 the modules were moved to 'org.eclipse.gemini.blueprint' and artifacts
+                // were renamed.
+                if (details.requested.group == 'org.eclipse.gemini.blueprint'
+                    && details.requested.version.startsWith('1.0')) {
```
Comment (tgeens) [inline build.gradle]: @vierbergenlars Is this better ?

Comment (vierbergenlars) [reply build.gradle]: Yes, I do think this looks better and more easy to understand the idea.

**Pull request merged**